gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/transformation/Transformation.java

```diff
@@ -59,66 +68,60 @@
    * @param vertexTransFunc     vertex transformation function
    * @param edgeTransFunc       edge transformation function
    */
-  public Transformation(TransformationFunction<GraphHead> graphHeadTransFunc,
-    TransformationFunction<Vertex> vertexTransFunc,
-    TransformationFunction<Edge> edgeTransFunc) {
+  public Transformation(
+    TransformationFunction<G> graphHeadTransFunc,
+    TransformationFunction<V> vertexTransFunc,
+    TransformationFunction<E> edgeTransFunc) {
 
-    if (graphHeadTransFunc == null &&
-      vertexTransFunc == null &&
-      edgeTransFunc == null) {
-      throw new IllegalArgumentException(
-        "Provide at least one transformation function.");
+    if (graphHeadTransFunc == null && vertexTransFunc == null && edgeTransFunc == null) {
+      throw new IllegalArgumentException("Provide at least one transformation function.");
     }
     this.graphHeadTransFunc = graphHeadTransFunc;
     this.vertexTransFunc    = vertexTransFunc;
     this.edgeTransFunc      = edgeTransFunc;
   }
 
   @Override
-  public LogicalGraph execute(LogicalGraph graph) {
+  public LG execute(LG graph) {
     return executeInternal(
       graph.getGraphHead(),
       graph.getVertices(),
       graph.getEdges(),
-      graph.getConfig());
+      graph.getFactory());
   }
 
   /**
    * Applies the transformation functions on the given datasets.
    *
-   * @param graphHeads  graph heads
-   * @param vertices    vertices
-   * @param edges       edges
-   * @param config      gradoop flink config
+   * @param graphHeads graph heads
+   * @param vertices vertices
+   * @param edges edges
+   * @param factory the factory that is responsible for creating an instance of the logical graph
    * @return transformed logical graph
    */
   @SuppressWarnings("unchecked")
-  protected LogicalGraph executeInternal(DataSet<GraphHead> graphHeads,
-    DataSet<Vertex> vertices, DataSet<Edge> edges, GradoopFlinkConfig config) {
+  protected LG executeInternal(DataSet<G> graphHeads, DataSet<V> vertices, DataSet<E> edges,
+    BaseGraphFactory<G, V, E, LG> factory) {
 
-    DataSet<GraphHead> transformedGraphHeads = graphHeadTransFunc != null ?
-      graphHeads.map(new TransformGraphHead(
-        graphHeadTransFunc, config.getGraphHeadFactory()))
-        .returns(TypeExtractor.createTypeInfo(
-          config.getGraphHeadFactory().getType())) : graphHeads;
+    DataSet<G> transformedGraphHeads = graphHeadTransFunc != null ?
```

Comment:
Kinda strange code style here, try:

```
    DataSet<G> transformedGraphHeads = graphHeadTransFunc != null ? graphHeads
      .map(new TransformGraphHead(graphHeadTransFunc, factory.getGraphHeadFactory()))
      .returns(TypeExtractor.createTypeInfo(factory.getGraphHeadFactory().getType())) : graphHeads;
```

Reply:
I like it strange .... OK - I'll change it.